src/widget/controlwidgetconnection.cpp

```diff
@@ -97,40 +97,67 @@ void ControlParameterWidgetConnection::setControlParameterUp(double v) {
 }
 
 ControlWidgetPropertyConnection::ControlWidgetPropertyConnection(
-        WBaseWidget* pBaseWidget, const ConfigKey& key,
-        ValueTransformer* pTransformer, const QString& propertyName)
+        WBaseWidget* pBaseWidget,
+        const ConfigKey& key,
+        ValueTransformer* pTransformer,
+        const QString& propertyName)
         : ControlWidgetConnection(pBaseWidget, key, pTransformer),
-          m_propertyName(propertyName.toLatin1()) {
+          m_propertyName(propertyName) {
+    QWidget* pWidget = pBaseWidget->toQWidget();
+    VERIFY_OR_DEBUG_ASSERT(pWidget) {
+        return;
+    }
+    const QMetaObject* meta = pWidget->metaObject();
+    VERIFY_OR_DEBUG_ASSERT(!propertyName.isEmpty() && meta) {
```

Comment:
Please split the unrelated conditional expressions into two separate code paths. The property name could be checked when entering this function before doing anything else.